modules/renter/upload_test.go

```diff
@@ -0,0 +1,33 @@
+package renter
+
+import (
+	"testing"
+)
+
+// TestRenterSiapathValidate verifies that the validateSiapath function correctly validates SiaPaths.
+func TestRenterSiapathValidate(t *testing.T) {
+	var pathtests = []struct {
+		in    string
+		valid bool
+	}{
+		{"valid/siapath", true},
+		{"../../../directory/traversal", false},
+		{"testpath", true},
+		{"valid/siapath/../with/directory/traversal", false},
```

Comment:
I don't think it's all that convenient anyway. Why would a user specify such a path? Personally I only use `..` when I want to reference a directory relative to the one I'm in now, but that doesn't apply to siapaths.